cpp/oneapi/dal/detail/array_utils.hpp

```diff
@@ -39,8 +39,47 @@ class array_via_policy {
 #endif
 };
 
+template <typename T, typename Body>
+inline auto dispath_by_policy(const array<T>& data, Body&& body) {
```

Comment:
`data` is used just to obtain an optional on queue. Isn't it better to pass `std::optional<sycl::queue>` instead?

Reply:
The problem is that `array::get_queue()` is not available for non-DPC++ case. That's why call of this method is under macro `ONEDAL_DATA_PARALLEL`

Reply:
I've tried `__IF_QUEUE__`, it looks worse than `dispatch_by_policy` because requires some code duplication for the `__IF_QUEUE__` and `__IF_NO_QUEUE__` cases.